core/src/avm2/object/xml_object.rs

```diff
@@ -365,6 +365,15 @@ impl<'gc> TObject<'gc> for XmlObject<'gc> {
         self.0.read().base.has_own_dynamic_property(name)
     }
 
+    fn has_property_via_in(
+        self,
+        activation: &mut Activation<'_, 'gc>,
+        name: &Multiname<'gc>,
+    ) -> Result<bool, Error<'gc>> {
+        let multiname = handle_input_multiname(name.clone(), activation);
+        Ok(self.has_own_property(&multiname))
```

Comment:
This doesn't seem correct. `in` needs to check whole chain. Consider `"propertyIsEnumerable" in xml` vs `xml.hasOwnProperty("propertyIsEnumerable")`.

Reply:
Would changing to `self.has_property` be enough?

Reply:
> This doesn't seem correct. `in` needs to check whole chain. Consider `"propertyIsEnumerable" in xml` vs `xml.hasOwnProperty("propertyIsEnumerable")`.

At the moment `"propertyIsEnumerable" in xml` seems to return `false` on master too.

Reply:
> Would changing to `self.has_property` be enough, like how `has_property_via_in` does by default?

I think that might work? I think it should end up calling the XML's `has_own_property`.

> At the moment `"propertyIsEnumerable" in xml` seems to return `false` on master too.

It's `true` in Flash.